src/textual/widgets/_radio_set.py

```diff
@@ -0,0 +1,148 @@
+"""Provides a RadioSet widget, which groups radio buttons."""
+
+from __future__ import annotations
+
+from typing import cast
+
+from ..containers import Container
+from ..css.query import DOMQuery, QueryError
+from ..message import Message
+from ._radio_button import RadioButton
+
+
+class RadioSet(Container):
+    """Widget for grouping a collection of radio buttons into a set.
+
+    When a collection of [RadioButton][textual.widgets.RadioButton]s are
+    grouped with this widget, they will be treated as a mutually-exclusive
+    grouping. If one button is turned on, the previously-on button will be
+    turned off.
+    """
+
+    DEFAULT_CSS = """
+    RadioSet {
+        border: round #666;
+        height: auto;
+        width: auto;
+    }
+
+    App.-light-mode RadioSet {
+        border: round #CCC;
+    }
+    """
+
+    def __init__(
+        self,
+        *buttons: str | RadioButton,
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+        disabled: bool = False,
+    ) -> None:
+        """Initialise the radio set.
+
+        Args:
+            buttons: A collection of labels or [RadioButton][textual.widgets.RadioButton]s to group together.
+            name: The name of the radio set.
+            id: The ID of the radio set in the DOM.
+            classes: The CSS classes of the radio set.
+            disabled: Whether the radio set is disabled or not.
+
+        Note:
+            When a `str` label is provided, a
+            [RadioButton][textual.widgets.RadioButton] will be created from
+            it.
+        """
+        super().__init__(
+            *[
+                (button if isinstance(button, RadioButton) else RadioButton(button))
+                for button in buttons
+            ],
+            name=name,
+            id=id,
+            classes=classes,
+            disabled=disabled,
+        )
+
+    @property
+    def _buttons(self) -> DOMQuery[RadioButton]:
+        """The buttons within the set."""
+        return cast(DOMQuery[RadioButton], self.query(RadioButton))
+
+    def on_mount(self) -> None:
+        """Perform some processing once mounted in the DOM."""
+        # It's possible for the user to pass in a collection of radio
+        # buttons, with more than one set to on. So here we check for that
+        # and, for want of a better approach, we keep the first one on and
+        # turn all the others off.
+        switched_on = self._buttons.filter(".-on")
+        if len(switched_on) > 1:
+            with self.prevent(RadioButton.Changed):
+                for button in switched_on[1:]:
+                    button.value = False
+
+    class Changed(Message, bubble=True):
```

Comment:
My preference is to keep child classes close to where they're used, but sure, happy to do that.

Reply:
I only suggested this because we seem to be listing all widget messages in the beginning and I appreciate the consistency.
If you have a personal preference for this way, I won't say my personal preference is better than yours! I thought the class was here just because it “happened” to be here.